pbf/views.py

```diff
@@ -848,6 +859,25 @@ def change_spirit_specific_resource(request, player_id, amount):
     # because some of them can change simultaneously with energy (e.g. Rot).
     return with_log_trigger(render(request, 'energy.html', {'player': player}))
 
+def gain_rot(request, player_id):
+    player = get_object_or_404(GamePlayer, pk=player_id)
+    player.spirit_specific_resource += player.rot_gain()
+    player.spirit_specific_per_turn_flags |= GamePlayer.ROT_GAINED_THIS_TURN
```

Comment:
Could technically check to make sure `ROT_GAINED_THIS_TURN` flag isn't already set; but, it's not like I expect people to be hacking their games by using the API URLs directly...

Reply:
If someone wanted to intentionally cheat, they always have the +1 button available to them and there's nothing that can be done to stop them from using, so in my eyes the big reason to check whether the flag is already set would be to prevent accidental double-clicks... but I also just tried and I can't actually get that to happen either. It's definitely something to consider, but I'm also looking at the fact that the gain/pay energy endpoints don't check their flags either.

There is also a broader discussion here that if someone is cheating in this game, there are social consequences since this is a co-op game.

The site is in general taking a pretty lax approach though, since anyone who knows the UUID of a game can go in and click on buttons as they like, so it seems to generally rely on trusting people to do what's right. I hope there doesn't come a day when that has to change.